src/osemosys2iamc/resultify.py

```diff
@@ -26,9 +26,40 @@
 import matplotlib.dates as mdates
 
 
-def read_file(filename) -> pd.DataFrame:
+def read_file(in_path: str, osemosys_param: str, region_name_option: str) -> pd.DataFrame:
+    """Reads in selected csv file as given in the config file into a Pandas DataFrame
 
+    Also, at the time of DataFrame creation, this function checks how it should derive
+    the name of the region to be used
+
+    Options:
```

Comment:
Please use the docstrings standard implemented in the rest of the package.  Replace `options:` with 
```python
Arguments
---------
```
Also please use underscores and not hyphens for your argument names.